sklearn/utils/estimator_checks.py

```diff
@@ -1721,3 +1722,25 @@ def check_classifiers_regression_target(name, Estimator):
     e = Estimator()
     msg = 'Unknown label type: '
     assert_raises_regex(ValueError, msg, e.fit, X, y)
+
+
+@ignore_warnings(category=DeprecationWarning)
+def check_decision_proba_consistency(name, Estimator):
+    # Check whether an estimator having both decision_function and
+    # predict_proba methods has outputs with perfect rank correlation.
+
+    rnd = np.random.RandomState(0)
+    X_train = np.random.randint(2, size=(10, 4))
+    y = np.random.randint(2, size=10)
+    estimator = Estimator()
+
+    set_testing_parameters(estimator)
+
+    if (hasattr(estimator, "decision_function") and
+            hasattr(estimator, "predict_proba")):
+
+        estimator.fit(X_train, y)
+        X_test = np.random.randint(2, size=(5, 4))
```

Comment:
A larger number of samples provides more discrimination. 10 here too wouldn't hurt.